### ISSUE TITLE
[Remote Translog] Add Tests for Remote Translog Transfer Snapshot

### ISSUE THREAD

**Issue by Bukhtawar** [opensearch-project/OpenSearch#5259]
**Is your feature request related to a problem? Please describe.**
To unblock the follow up PRs I would defer tests in the PR https://github.com/opensearch-project/OpenSearch/pull/4480 to another PR
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
